workspaces/bulk-import/plugins/bulk-import-backend/src/gitlab/gitlabApiService.ts

```diff
@@ -758,4 +758,139 @@ export class GitlabApiService {
       },
     );
   }
+
+  async getPullRequest(
+    repoUrl: string,
+    pullRequestNumber: number,
+  ): Promise<{
+    title?: string;
+    body?: string;
+    merged?: boolean;
+    lastUpdated?: string;
+    prSha?: string;
+    prBranch?: string;
+  }> {
+    const glConfig = this.integrations.gitlab.byUrl(repoUrl)?.config;
+    if (!glConfig) {
+      this.logger.error(
+        `No Gitlab integration config found for repo ${repoUrl}`,
+      );
+      return {};
+    }
+
+    const gitUrl = gitUrlParse(repoUrl);
+    const owner = gitUrl.owner;
+    const repo = gitUrl.name;
+
+    const credentials = await this.gitlabCredentialsProvider.getAllCredentials({
+      host: glConfig.host,
+    });
+
+    if (credentials.length === 0) {
+      this.logger.error(`No credentials for GL integration`);
+      return {};
+    }
+
+    for (const credential of credentials) {
```

Comment:
@AndrienkoAleksandr just to clarify, is this to ensure the first working credential will return the fetched data?

Reply:
Yes, it is .